scenario/utils.ts

```diff
@@ -254,3 +267,199 @@ export async function fetchLogs(
     return contract.queryFilter(filter, fromBlock, toBlock);
   }
 }
+
+export async function executeOpenProposal(
+  dm: DeploymentManager,
+  { id, startBlock, endBlock }: OpenProposal
+) {
+  const governor = await dm.contract('governor');
+
+  if (!governor) {
+    throw new Error("cannot execute proposal without governor");
+  }
+
+  const blockNow = await dm.hre.ethers.provider.getBlockNumber();
+  const blocksUntilStart = startBlock - blockNow;
+  const blocksUntilEnd = endBlock - Math.max(startBlock, blockNow);
+
+  if (blocksUntilStart > 0) {
+    await mineBlocks(dm, blocksUntilStart);
+  }
+
+  if (blocksUntilEnd > 0) {
+    for (const whale of COMP_WHALES) {
+      try {
+        // Voting can fail if voter has already voted
+        const voter = await impersonateAddress(dm, whale);
+        await setNextBaseFeeToZero(dm);
+        await governor.connect(voter).castVote(id, 1, { gasPrice: 0 });
+      } catch (err) {
+        debug(`Error while voting for ${whale}`, err.message);
+      }
+    }
+    await mineBlocks(dm, blocksUntilEnd);
+  }
+
+  // Queue proposal (maybe)
+  const state = await governor.state(id);
+  if (state == ProposalState.Succeeded) {
+    await setNextBaseFeeToZero(dm);
+    await governor.queue(id, { gasPrice: 0 });
+  }
+
+  const proposal = await governor.proposals(id);
+  await setNextBlockTimestamp(dm, proposal.eta.toNumber() + 1);
+
+  // Execute proposal (w/ gas limit so we see if exec reverts, not a gas estimation error)
+  await setNextBaseFeeToZero(dm);
+  await governor.execute(id, { gasPrice: 0, gasLimit: 12000000 });
+}
+
+// Instantly executes some actions through the governance proposal process
+export async function fastGovernanceExecute(
+  dm: DeploymentManager,
+  proposer: SignerWithAddress,
+  targets: string[],
+  values: BigNumberish[],
+  signatures: string[],
+  calldatas: string[]
+) {
+  const governor = await dm.contract('governor');
+
+  if (!governor) {
+    throw new Error("cannot create governance proposal without governor");
+  }
+
+  await setNextBaseFeeToZero(dm);
+
+  const proposeTxn = await (
+    await governor.connect(proposer).propose(
+      targets,
+      values,
+      signatures,
+      calldatas,
+      'FastExecuteProposal',
+      { gasPrice: 0 }
+    )
+  ).wait();
+  const proposeEvent = proposeTxn.events.find(event => event.event === 'ProposalCreated');
+  const [id, , , , , , startBlock, endBlock] = proposeEvent.args;
+
+  await executeOpenProposal(dm, { id, startBlock, endBlock });
+}
+
+async function relayMumbaiMessage(
+  governanceDeploymentManager: DeploymentManager,
+  bridgeDeploymentManager: DeploymentManager,
+) {
+  const STATE_SENDER = '0xeaa852323826c71cd7920c3b4c007184234c3945';
+  const MUMBAI_RECEIVER_ADDRESSS = '0x0000000000000000000000000000000000001001';
```

Comment:
Could these be dervied via `aliases` as well?

Reply:
yes for `STATE_SENDER`.

I added `fxRoot` to Goerli roots, with a relation to go from fxRoot -> stateSender

`MUMBAI_RECEIVER_ADDRESS` isn't crawlable, however. it's just an address that is hard-coded into `fxChild`, unfortunately